odl/tomo/operators/spect_trafo.py

```diff
@@ -0,0 +1,226 @@
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+"""Single-photon emission computerized tomography (SPECT) projectors """
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+from builtins import super
+
+# External
+try:
+    from NiftyPy.NiftyRec import (SPECT_project_parallelholes,
+                                  SPECT_backproject_parallelholes)
+    NIFTYREC_AVAILABLE = True
+except ImportError:
+    NIFTYREC_AVAILABLE = False
+
+import numpy as np
+
+# Internal
+import odl
+from odl.operator.operator import Operator
+
+
+_SUPPORTED_IMPL = ('niftyrec')
+__all__ = ('SpectProject', 'SpectBackproject')
+
+
+class SpectProject(Operator):
+    """Spect projector using backend NiftyRec"""
+    def __init__(self, dom, geometry, attenuation, psf, use_gpu=False,
+                 impl='niftyrec', **kwargs):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        dom : `DiscreteLp`
+            Discretized space, the domain of the forward projector
+        geometry : `Geometry`
+            The geometry of the transform. An ODL geometry instance
+            that contains information about the operator domain
+        attenuation : `DiscreteLpVector`
+            Linear attenuation map for SPECT.
+        psf : `DiscreteLpVector`
+            Point Spread Functions for the correction of the
+            collimator-detector response
+        use_gpu : `bool`, optional
+            'False' if not using GPU
+            `True` if using GPU provided that `impl` supports it
+        impl='niftyrec', optional
+            Implementation back-end for the transform. Supported back-ends:
+            NiftyRec Tomography Toolbox through NiftyPy a Python wrapper
+            for NiftyRec
+        """
+        impl, impl_in = str(impl).lower(), impl
+        if impl not in _SUPPORTED_IMPL:
+            raise ValueError('implementation {!r} not supported.'
+                             ''.format(impl_in))
+
+        self.geometry = geometry
+        det_nx = geometry.det_grid.shape[0]
+        det_ny = geometry.det_grid.shape[1]
+        startAngle = float(geometry.motion_grid.points()[0])
+        stopAngle = float(geometry.motion_grid.points()[-1]) * (180 / np.pi)
+        N_proj = len(geometry.motion_grid.points())
```

Comment:
Another general thing: If you call a method which is not a property, i.e. you use parentheses like for `grid.points()`, you need to assume that it comes at a computational cost. So calling it three times just to pick out one value adds unnecessary overhead (not so much for 1d angles, but anyway).
If you see yourself using the result of such a method call more than once, it may be a good idea to store the result. See also the use of `corners` below.